### DISCUSSION TITLE
Gradient Fill Element

### DISCUSSION THREAD

**Discussion by RossP4ge** [ExperienceLovelace/ha-floorplan#148] [Q&A] [answered]
Hey, I'm trying to figure out how to change the colour of the gradient of an SVG object:

![image](https://user-images.githubusercontent.com/87759452/134589829-f98e8233-4ad2-420b-96f3-0b4b9a3a3bd8.png)

This is the kind of thing I want (just edited gradient in inkspace)

![image](https://user-images.githubusercontent.com/87759452/134590333-775e6693-3aef-4b9c-a0b6-3c20fe42fdc2.png)

The white circle is surrounded by a blue gradient in the picture above. I would like to sync the gradient with the colour of the white circle, my yaml rule for this looks like so:

                - entity: switch.emily_s_lamp
                  tap_action: homeassistant.toggle
                  state_action:
                    - action: call-service
                      service: floorplan.class_set
                      element: switch.emily_s_lamp
                      service_data: 
                        class: '${(entity.state === "on") ? "lamp-on" : "lamp-off"}'
                    - action: call-service
                      service: floorplan.class_set
                      element: switch.emily_s_lamp.gradient
                      service_data: 
                        class: '${(entity.state === "on") ? "lamp-gradient-on" : "lamp-gradient-off"}'

CSS:

.lamp-on, .lamp-on * {
  fill: #FF932C !important;
}

.lamp-off, .lamp-off * {
  fill: #D3D3D3 !important;
}

.lamp-graident-on, .lamp-gradient-on * {
  **NOT SURE WHAT TO PUT** !important;
}

.lamp-gradient-off, .lamp-gradient-off * {
  **NOT SURE WHAT TO PUT** !important;
}

I have tried this:  fill: radial-gradient(rgba(211,211,211,1) 0%, rgba(211,211,211,0.05) 100%, but it didn't work unsurprisingly.

I am aware of the lights example, but I really couldn't understand the example (as I have next to no coding knowledge) and it wasn't in a CSS format.

Hopefully i've explained it fairly well. Any help would be greatly appreciated.

**Comment by OzGav** [comment]
I don’t do this but the light example shows this in the YAML so give this a go

fill: rgb(211,211,211]; stop-color: rgb(211,211,211]; stop-opacity: 1;

Colours need to be adjusted and I think you can replace the rgb(211,211,211] with just your hex code. Eg #FF932C

**Reply by RossP4ge** [reply]
> I don’t do this but the light example shows this in the YAML so give this a go
> 
> fill: rgb(211,211,211]; stop-color: rgb(211,211,211]; stop-opacity: 1;
> 
> Colours need to be adjusted and I think you can replace the rgb(211,211,211] with just your hex code. Eg #FF932C

That didn't work unfortunately, it just didn't recognize. I've gone around it a different way, but i'd be really interested if such a CSS block could do what I'm after. I just set the gradient to the colour I wanted and toggled opacity. 

Here is the associated yaml and css (I imagine it could be cleaner!):

# Fixed Colour Lamp 
                - entity: switch.emily_s_lamp
                  tap_action: homeassistant.toggle
                  state_action:
                    - action: call-service
                      service: floorplan.class_set
                      element: switch.emily_s_lamp
                      service_data: 
                        class: '${(entity.state === "on") ? "lamp-on" : "lamp-off"}'
                - element: switch.emily_s_lamp.gradient  
                  entity: switch.emily_s_lamp
                  tap_action: homeassistant.toggle
                  state_action:
                    - action: call-service
                      service: floorplan.class_set
                      state: switch.emily_s_lamp
                      element: switch.emily_s_lamp.gradient
                      service_data: 
                        class: '${(entity.state === "on") ? "lamp-gradient-on" : "lamp-gradient-off"}'

.lamp-on, .lamp-on * {
  fill: #ffd6aa !important;
}

.lamp-off, .lamp-off * {
  fill: #D3D3D3 !important;
  fill-opacity: 0.5 !important; 
}

.lamp-gradient-on, .lamp-gradient-on * {
  opacity: 1 !important;
}

.lamp-gradient-off, .lamp-gradient-off * {
  opacity: 0 !important;
}

On:
![image](https://user-images.githubusercontent.com/87759452/134699133-7c009e63-16a7-40e0-9691-15b175d0ffb2.png)

Off:
![image](https://user-images.githubusercontent.com/87759452/134699157-0307fb9b-f7f1-4f7e-be68-c9a0a4b5ede6.png)

**Reply by OzGav** [reply]
Glad you have it working and sorry I couldn’t be much more help. Would be great if you could mark your solution as the answer.

**Comment by RossP4ge** [comment, marked as answer]
Solution: https://github.com/ExperienceLovelace/ha-floorplan/discussions/148#discussioncomment-1380497
